### ISSUE TITLE
Oprofile/Perf Support

### ISSUE THREAD

**Issue by gecko656** [LegacyXperia/local_manifests#314]
Hello,

for my current project I need to profile native libraries. To be able to do that the linux kernel provided profiling tools oprofile and perf would be very helpful. Maybe some of other developers can also benefit of that tools, because they also allow to profile the kernel libs. I already tried to compile the current with cm 10.2, but I'm not experienced in kernel compilation. The compilation with the current defconfig (nAa_hallon_defconfig) succeeded, but when I add the addtion config-flags that should enable oprofiling in this defconfig, nothing changes. The kernel seems to be the same as without the oprofile flags. The normal way to start oprofile is to run for example opcontrol --setup in the adb shell, which fails at the moment:
Cannot create directory /dev/oprofile: Permission dendied
Cannot create directory /data/oprofile: Permission denied
Cannot create directory /data/oprofile/samples: No such file or directoy
It would be very helpful, if somebody can help me compiling the kernel with enabled oprofile. Or if somebody already has a kernel that supports oprofile it would also very helful.
The changes I made to the nAa_hallon_defconfig were
CONFIG_PROFILING=y
CONFIG_OPROFILE=y
CONFIG_HAVE_OPROFILE=y
I compiled with the help of mikes tutorial on xda. ( http://forum.xda-developers.com/showthread.php?t=2397249 )
Thanks,
gecko

**Comment by mikeNG** [comment]
I will look into this and might enable oprofile in next build. I cannot test it so you will have to test if it works.

**Comment by gecko656** [comment]
okay, that would be very nice. Thanks a lot for your help! just let me know if I can help you.

**Comment by luisbandalap** [comment]
@gecko656 did you try adding/modifying this lines in your kernel config file?

CONFIG_HW_PERF_EVENTS=y
CONFIG_PERF_EVENTS=y

http://lockephan.wordpress.com/2012/03/03/how-to-enable-oprofile-on-ics-4-0-3/
http://lockephan.wordpress.com/2012/03/03/how-to-enable-oprofile-on-ics-4-0-3/

**Comment by gecko656** [comment]
no i didn't used those lines, yet. as mike said he wants to try it for his next build, I decided to switch to CM11.0, where I'm struggling with the standard compiling at the moment. But after the first successful CM11-build I will try it out. Thanks for your help, I will report if it has worked!

**Comment by gecko656** [comment]
I got the compilation of CM11.0 working. I've tried your @luisbandalap changes, but the OProfile does still not work. (see error message below) In addition I tried to with
CONFIG_OPROFILE_ARMV7=y
and I also tried to compile OProfile as a module. Moreover I tried some other changes that I've found in the web like the following, but also with no success.
https://wiki.mozilla.org/Mobile/Fennec/Android/Profiling#Using_oprofile_on_Galaxy_Nexus
When i tried that command:
adb shell dmesg | grep oprofile
I get the output that OProfile is running on an ARMv7 Snapdragon. But when I want to run opcontrol --quick or opcontrol --setup I now get the following error:

oprofile: "too many events passed. Failed, oprofile returned exit code: 256"

So all in all I think it works somehow better now, but still I can't start profiling. 
@mikeNG any idea where the problem could be?
thanks @luisbandalap for your help. Any other suggestions what I can do?

**Comment by mikeNG** [comment]
I enabled profiling stuff in the next cm-10.2/cm-11.0 builds

**Comment by gecko656** [comment]
Oh, thanks a lot for your help!  I saw your changes in the defconfigs. Did you do something else? I just want to learn something.
CONFIG_PROFILING=y
CONFIG_OPROFILE=m
CONFIG_HAVE_OPROFILE=y
CONFIG_HW_PERF_EVENTS=y
CONFIG_PERF_EVENTS=y

Concerning PERF. If anyone else wants to use it:
I just enabled the PERF-Flags in the defconfig and compiled the ROM and kernel with
lunch cm_hallon-eng
make bacon
Afterwards it can be used within the console. I'm not a dev and I think this is not the best way to get it working, but it actually works.
